sulong/projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/LLVMContext.java

```diff
@@ -212,6 +223,11 @@ public LLVMContext(LLVMLanguage language, Env env, Configuration activeConfigura
         } else {
             tracer = null;
         }
+        this.condStorage = new ConcurrentHashMap<>();
```

Comment:
Why did you choose `ConcurrentHashMap` over other concurrent collections or implementing your own? Just curious.

Reply:
because there might be multiple threads accessing this collection at once, so just to be sure i used a concurrent collection, not sure if that's needed in my case tho. hash-map because i want to use the (mutex/thread/...)-id to get the corresponding mutex/thread/...

Reply:
You definitely need to synchronize access to the map if it is accessed from multiple threads. The `ConcurrentHashMap` is a good choice for this (see, e.g., [here](https://crunchify.com/hashmap-vs-concurrenthashmap-vs-synchronizedmap-how-a-hashmap-can-be-synchronized-in-java/)).